### ISSUE TITLE
Unable to explicitly import aliased interfaces

### ISSUE THREAD

**Issue by infiloop2** [ethereum/solidity#13564]
## Description

Explicitly importing an interface which is defined as an aliased interface fails. Importing the whole file works.

## Environment

- Compiler version: `solc --version` gives `0.8.6+commit.11564f7e.Darwin.appleclang`
- Target EVM version (as per compiler settings): 
- Framework/IDE (e.g. Truffle or Remix): VSCode
- EVM execution environment / backend / blockchain client:
- Operating system: MacOS Monterey 12.6

## Steps to Reproduce

Create 4 files

InterfaceA.sol
```
pragma solidity ^0.8.0;

interface InterfaceA {
    function check() external view;
}

```

AliasInterface.sol
```
pragma solidity ^0.8.0;
import {InterfaceA as AliasInterface} from "./InterfaceA.sol";
```

Contract1.sol
```
pragma solidity 0.8.6;
import {AliasInterface} from "./AliasInterface.sol";

contract Contract1 is AliasInterface {
    function check() external view override {}
}
```

Contract2.sol
```
pragma solidity 0.8.6;
import "./AliasInterface.sol";

contract Contract2 is AliasInterface {
    function check() external override view {}
}

```

Note that Contract2 compiles (`solc contract2.sol`) whereas Contract1 gives an error (`solc contract1.sol`).

**Comment by naveed949** [comment]
@infiloop2 in this case for Contract1.sol it should work like this:
```
pragma solidity 0.8.6;
import {AliasInterface} from "./AliasInterface.sol";

contract Contract1 is InterfaceA {
    function check() external view override {}
}
```

Reason of this is because **alias** works as kind of macro, So it loads/injects actual Interface (InterfaceA) in contract1 when you explicitly import it. Thus at the time when you're using it like `contract Contract1 is AliasInterface {` there's no alias exists (because it already got resolved above into InterfaceA)

**Comment by infiloop2** [comment]
@naveed949 I understand the underlying technical issue, but from a user perspective without knowing the internal details of `AliasInterface.sol`, the fact that `Contract2` compiles whereas `Contract1` doesn't is very counter intuitive

**Comment by cameel** [comment]
Duplicate of #11855.

This was indeed a bug but has already been fixed in 0.8.8.

By the way, please note that using 0.8.8 is not recommended due to a bug in the storage layout of user-defined value types, which was immediately fixed in 0.8.9. When upgrading to a version that's not the latest please consider the [list of known bugs](https://github.com/ethereum/solidity/blob/develop/docs/bugs_by_version.json).
